pkg/otel/arrow_record/consumer.go

```diff
@@ -221,13 +335,15 @@ func (c *Consumer) Consume(bar *colarspb.BatchArrowRecords) ([]*record_message.R
 
 		sc.bufReader.Reset(payload.Record)
 		if sc.ipcReader == nil {
+			c.schemaResetCounter.Add(ctx, 1, metric.WithAttributes(attribute.String("payload_type", payload.Type.String()), c.uniqueAttr))
 			ipcReader, err := ipc.NewReader(
 				sc.bufReader,
-				ipc.WithAllocator(common.NewLimitedAllocator(memory.NewGoAllocator(), c.config.memLimit)),
+				ipc.WithAllocator(c.allocator),
```

Comment:
Not strictly equivalent to the previous code. With this modification we have a single allocator for all the payloads instead of one allocator per payload type. It's probably ok but that could explain why we observe a different behavior with the previous approach.